test/pipelinerun_test.go

```diff
@@ -160,14 +161,36 @@ func TestPipelineRun(t *testing.T) {
 				if !r.Status.GetCondition(duckv1alpha1.ConditionSucceeded).IsTrue() {
 					t.Fatalf("Expected TaskRun %s to have succeeded but Status is %v", taskRunName, r.Status)
 				}
-				for name, key := range map[string]string{
-					pipelineRunName: pipeline.PipelineRunLabelKey,
-					pipelineName:    pipeline.PipelineLabelKey,
-				} {
-					expectedName := getName(name, i)
-					lbl := pipeline.GroupName + key
-					if val := r.Labels[lbl]; val != expectedName {
-						t.Errorf("Expected label %s=%s but got %s", lbl, expectedName, val)
+
+				// Check label propagation to Tasks
+				labels := make(map[string]string, len(pr.ObjectMeta.Labels)+3)
+				for key, val := range pr.ObjectMeta.Labels {
+					labels[key] = val
+				}
+				labels[pipeline.GroupName+pipeline.PipelineLabelKey] = getName(pipelineName, i)
+				labels[pipeline.GroupName+pipeline.PipelineRunLabelKey] = getName(pipelineRunName, i)
```

Comment:
what is the significance of `pipeline.GroupName+pipeline.PipelineRunLabelKey` and `pipeline.GroupName+pipeline.PipelineLabelKey`? (i assume these correspond to some label generation logic in the controller - maybe a comment here would help?)

Reply:
Yeah those are static labels added by the controller, but definitely could use a comment.